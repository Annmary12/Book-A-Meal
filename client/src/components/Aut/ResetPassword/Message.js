import React from 'react';
import { Link } from 'react-router-dom';
import styles from './resetpassword.scss';

const Message = () => (
  <div className={styles.mailSent}>
    <h4>Reset password link would be sent to your email shortly.</h4>

    <p>
      Go back to <Link to="login">Login</Link>
    </p>
  </div>
);

export default Message;
