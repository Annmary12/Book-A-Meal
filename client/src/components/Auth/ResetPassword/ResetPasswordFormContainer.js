import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { resetPasswordSchema } from '../../../utils/validationSchemas.js';
import AuthForm from '../AuthForm.js';
import { sendResetMail } from '../../../actions/resetPassword.js';

const resetPasswordFormConfig = {
  mapPropsToValues: () => ({
    email: ''
  }),
  validationSchema: resetPasswordSchema,
  handleSubmit: async (values, { setSubmitting, props }) => {
    setSubmitting(true);
    await props.dispatch(sendResetMail(values));
    setSubmitting(false);
  }
};
const mapStateToProps = state => ({
  type: 'reset',
  mailSent: state.resetPassword.mailSent
});

export default compose(
  connect(mapStateToProps),
  withFormik(resetPasswordFormConfig)
)(AuthForm);
