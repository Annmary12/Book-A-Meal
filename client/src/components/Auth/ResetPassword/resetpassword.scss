@import '../../../../assets/scss/_variables';

.mailSent{
  text-align: center;
}