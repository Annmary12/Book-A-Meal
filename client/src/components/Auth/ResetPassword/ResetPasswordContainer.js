import { connect } from 'react-redux';
import ResetPassword from './ResetPassword';

const mapStateToProps = state => ({
  mailSent: state.resetPassword.mailSent
});

export default connect(mapStateToProps)(ResetPassword);
