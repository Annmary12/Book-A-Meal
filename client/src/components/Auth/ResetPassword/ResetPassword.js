import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import AuthArea from '../AuthArea.js';
import SentMessage from './SentMessage.js';

import styles from './resetpassword.scss';
import ResetPasswordFormContainer from './ResetPasswordFormContainer.js';

const ResetPassword = ({ mailSent }) => (
  <AuthArea>
    <div className={styles.mailSentCard}>
      {
        mailSent ?
          <SentMessage /> :
          <Fragment>
            <ResetPasswordFormContainer />
            <p style={{ textAlign: 'center' }}>
            Already have an account? <Link to="/login">Login</Link>
            </p>
          </Fragment>
      }
    </div>
  </AuthArea>
);

ResetPassword.propTypes = {
  mailSent: PropTypes.bool.isRequired
};


export default ResetPassword;
