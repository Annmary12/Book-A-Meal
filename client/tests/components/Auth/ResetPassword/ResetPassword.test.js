import React from 'react';
import ResetPassword from '../../../../src/components/Auth/ResetPassword/ResetPassword';

describe('ResetPassword component', () => {
  it('should match snapshot if mailSent is true', () => {
    const wrapper = shallow(<ResetPassword mailSent={true} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot if mailSent is false', () => {
    const wrapper = shallow(<ResetPassword mailSent={false} />);

    expect(wrapper).toMatchSnapshot();
  });
});