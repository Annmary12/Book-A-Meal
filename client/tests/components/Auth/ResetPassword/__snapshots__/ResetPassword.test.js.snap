// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword component should match snapshot if mailSent is false 1`] = `
<AuthArea>
  <div
    className="mailSentCard"
  >
    <Connect(C) />
    <p
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Already have an account? 
      <Link
        replace={false}
        to="/login"
      >
        Login
      </Link>
    </p>
  </div>
</AuthArea>
`;

exports[`ResetPassword component should match snapshot if mailSent is true 1`] = `
<AuthArea>
  <div
    className="mailSentCard"
  >
    <SentMessage />
  </div>
</AuthArea>
`;
